packages/shared/hooks/useRefAutoFocus/useRefAutoFocus.ts

```diff
@@ -0,0 +1,20 @@
+import { DependencyList, MutableRefObject, useEffect, useRef } from 'react';
+
+/**
+ * Returns `ref` object that is automatically focused when `canFocus` is `true`.
+ * Focus can be also re triggered by changing any of the `refocusDeps`.
+ */
+export function useRefAutoFocus<T extends { focus(): void }>(options: {
```

Comment:
I wasn't sure if `useRefAutoFocus` is the right name but I guess it really is an autofocus _if_ `canFocus` is true – it'll call focus as soon as the component is rendered.